src/runners/plugins/slack.py

```diff
@@ -0,0 +1,142 @@
+import os
+import json
+
+from slackclient import SlackClient
+
+from runners.helpers import log
+from runners.helpers import db
+from runners.helpers import kms
+
+
+def message_template(vars):
+    payload = None
+
+    # if we have Slack user data, send it to template
+    if 'user' in vars:
+        params = {'alert': vars['alert'], 'properties': vars['properties'], 'user': vars['user']}
+    else:
+        params = {'alert': vars['alert'], 'properties': vars['properties']}
+
+    try:
+        # retrieve Slack message structure from javascript UDF
+        rows = db.connect_and_fetchall(
+            "select " + vars['template'] + "(parse_json('" + json.dumps(params) + "'))")
+        row = rows[1]
+
+        if len(row) > 0:
+            log.debug(f"Template {vars['template']}", ''.join(row[0]))
+            payload = json.loads(''.join(row[0]))
+        else:
+            log.error(f"Error loading javascript template {vars['template']}")
+            raise Exception("Error loading javascript template " + {vars['template']})
+    except Exception as e:
+        log.error(f"Error loading javascript template", e)
+        raise
+
+    return payload
+
+
+def record_status(response, alert_id):
+    query = f"UPDATE results.alerts SET handled=%s WHERE alert:ALERT_ID='{alert_id}'"
+    print('Updating alert table:', query)
+    try:
+
+        db.execute(query, params=str(response))
+    except Exception as e:
+        log.error(e, f"Failed to update alert {alert_id} with status {response}")
+
+
+def handle(alert_shell, type='slack', recipient_email=None, channel=None, template=None, message=None):

```

Comment:
what is `alert_shell`? why not just pass in `alert`? it's the only thing you use...

Reply:
alert_shell is all of the columns from the alerts table, which includes ALERT, CORRELATION_ID, HANDLED, and so on. It's best practice to pass the full shell into the handler so that e.g. if the handler wants to use the CORRELATION_ID or some other metadata from the alert shell, it's available.